java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/protocol/match/MatchField.java

```diff
@@ -31,15 +31,22 @@
 import org.projectfloodlight.openflow.types.VxlanNI;
 import org.projectfloodlight.openflow.types.VFI;
 
+import java.util.Set;
+import com.google.common.collect.ImmutableSet;
+
 public class MatchField<F extends OFValueType<F>> {
     private final String name;
     public final MatchFields id;
-    private final Prerequisite<?>[] prerequisites;
+    private final Set<Prerequisite<?>> prerequisites;
 
     private MatchField(final String name, final MatchFields id, Prerequisite<?>... prerequisites) {
         this.name = name;
         this.id = id;
-        this.prerequisites = prerequisites;
+        if (prerequisites == null) {
+            this.prerequisites = ImmutableSet.<Prerequisite<?>>of();
+        } else { 
+            this.prerequisites = ImmutableSet.<Prerequisite<?>>copyOf(prerequisites);
```

Comment:
Tiny stylistic nit - you should be able to drop the explicit `<Prerequisite<?>>` here...